subprojects/core/src/test/groovy/org/gradle/api/internal/tasks/properties/PropertyValidationAccessTest.groovy

```diff
@@ -126,7 +126,7 @@ class PropertyValidationAccessTest extends Specification {
 
         then:
         problems.keySet() == validationProblems(TaskWithIterableInIterable, [
-                "property 'beans<List><Set>.nonAnnotated' is not annotated with an input or output annotation",
+                "property 'beans<Set><NestedBean>.nonAnnotated' is not annotated with an input or output annotation",
```

Comment:
Looks better! But now it just looks weird that we don't have the `<List>` in front of those guys... It's kind of a code smell showing that we're thinking slightly wrong about this. WDYT?

Reply:
I used the `*` operator instead.